File: python/program/Elektronik.py
# Import class Toko
from Toko import Toko

# Class Elektronik (Child dari Toko)
class Elektronik(Toko):
    def __init__(self, id="", namaProduk="", harga=0,
                 kategori="", jumlahStok=0, statusBarang=""):
        super().__init__(id, namaProduk, harga)
        self.kategori = kategori
        self.jumlahStok = jumlahStok
        self.statusBarang = statusBarang

    # Getter & Setter
    def getKategori(self):
        return self.kategori

    def setKategori(self, kategori):
        self.kategori = kategori

    def getJumlahStok(self):
        return self.jumlahStok

    def setJumlahStok(self, jumlahStok):
        self.jumlahStok = jumlahStok

    def getStatusBarang(self):
        return self.statusBarang

    def setStatusBarang(self, statusBarang):
        self.statusBarang = statusBarang

File: php/program/Elektronik.php
<?php
require_once "Toko.php";

// Class Elektronik (Child dari Toko)
class Elektronik extends Toko {
    private $kategori;
    private $jumlahStok;
    private $statusBarang;

    // Constructor default
    public function __construct($id = "", $namaProduk = "", $harga = 0, $fotoProduk = "",
                                $kategori = "", $jumlahStok = 0, $statusBarang = "") {
        // Panggil constructor parent
        parent::__construct($id, $namaProduk, $harga, $fotoProduk);
        $this->kategori = $kategori;
        $this->jumlahStok = $jumlahStok;
        $this->statusBarang = $statusBarang;
    }

    // Getter & Setter untuk kategori
    public function getKategori() {
        return $this->kategori;
    }

    public function setKategori($kategori) {
        $this->kategori = $kategori;
    }

    // Getter & Setter untuk jumlah stok
    public function getJumlahStok() {
        return $this->jumlahStok;
    }

    public function setJumlahStok($jumlahStok) {
        $this->jumlahStok = $jumlahStok;
    }

    // Getter & Setter untuk status barang
    public function getStatusBarang() {
        return $this->statusBarang;
    }

    public function setStatusBarang($statusBarang) {
        $this->statusBarang = $statusBarang;
    }
}
?>
